### PR TITLE
[Fix] Discover supports private tenant

### PR DESCRIPTION
**Pull request by Desvelao**
Hi team,

this PR resolves:
- The Discover component uses `/elastic/esAlerts` and supports Private tennant in ODFE. This solves the table in Modules/Security Events.
- Added support when push the Refresh button to refresh the table

It is part of #2639

### CODE REVIEW

File: public/components/common/hooks/useResfreshAngularDiscover.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { useState, useEffect } from 'react';
+import { ModulesHelper } from '../modules/modules-helper';
+
+export function useRefreshAngularDiscover(): number {
+  const [refresh, setRefresh] = useState(Date.now());
+  useEffect(() => {
+    let subscription;
+    ModulesHelper.getDiscoverScope()
+      .then(scope => {
+        subscription = scope.$watchCollection('fetchStatus',
```
Comment (frankeros) [inline public/components/common/hooks/useResfreshAngularDiscover.ts:21]: why did you use $watchCollection instead of $watch? `fetchStatus` is a string

Comment (Desvelao) [reply public/components/common/hooks/useResfreshAngularDiscover.ts:21]: You are right :+1:

**Comment by Desvelao** [comment]
I replaced the `JSON.stringify` method to compare objects by `_isEqual` from `lodash` package.

**Review by jsanchez91 — APPROVED** [review]
LGTM!

**Pull request merged**